src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -1,202 +1,326 @@
-  public ElasticSearch()
-  {
-  }
-
-  public URI getUri()
-  {
-    return uri;
-  }
-
-  @Override
- public void validate() throws MimeTypeParseException {
-    new MimeType(this.mimeType);
-  }
-
-  /*
-   * We use URL for the setter as stapler can autoconvert a string to a URL but not to a URI
-   */
-  @DataBoundSetter
-  public void setUri(URL url) throws URISyntaxException
-  {
-    this.uri = url.toURI();
-  }
-
-  public void setUri(URI uri) throws URISyntaxException
-  {
-    this.uri = uri;
-  }
-
-  public String getUsername()
-  {
-    return username;
-  }
-
-  @DataBoundSetter
-  public void setUsername(String username)
-  {
-    this.username = username;
-  }
-
-  public String getPassword()
-  {
-    return Secret.toString(password);
-  }
-
-  @DataBoundSetter
-  public void setPassword(String password)
-  {
-    this.password = Secret.fromString(password);
-  }
-  
-  @DataBoundSetter
-  public void setMimeType(String mimeType) {
-    this.mimeType = mimeType;
-  }
-  
-  public String getMimeType() {
-    return mimeType;
-  }
-  
-  @Override
-  public boolean equals(Object obj)
-  {
-    if (obj == null)
-      return false;
-    if (this == obj)
-      return true;
-    if (getClass() != obj.getClass())
-      return false;
-    ElasticSearch other = (ElasticSearch) obj;
-    if (!Secret.toString(password).equals(other.getPassword()))
-    {
-      return false;
-    }
-    if (uri == null)
-    {
-      if (other.uri != null)
-        return false;
-    }
-    else if (!uri.equals(other.uri))
-    {
-      return false;
-    }
-    if (username == null)
-    {
-      if (other.username != null)
-        return false;
-    }
-    else if (!username.equals(other.username))
-    {
-      return false;
-    }
-    else if (!mimeType.equals(other.mimeType))
-    {
-      return false;
-    }
-    return true;
-  }
-
-  @Override
-  public int hashCode()
-  {
-    final int prime = 31;
-    int result = super.hashCode();
-    result = prime * result + ((uri == null) ? 0 : uri.hashCode());
-    result = prime * result + ((username == null) ? 0 : username.hashCode());
-    result = prime * result + Secret.toString(password).hashCode();
-    return result;
-  }
-
-  @Override
-  public ElasticSearchDao createIndexerInstance()
-  {
-    ElasticSearchDao esDao = new ElasticSearchDao(getUri(), username, Secret.toString(password));
-    esDao.setMimeType(getMimeType());
-    return esDao;
-  }
-
-  @Extension
-  public static class ElasticSearchDescriptor extends LogstashIndexerDescriptor
-  {
     @Override
-    public String getDisplayName()
-    {
-      return "Elastic Search";
+    public int hashCode() {
+        final int prime = 31;
+        int result = super.hashCode();
+        result = prime * result + ((uri == null) ? 0 : uri.hashCode());
+        result = prime * result + ((username == null) ? 0 : username.hashCode());
+        result = prime * result + Secret.toString(password).hashCode();
+        return result;
```

Comment:
customServerCertificateId should be included in the equals method.

Reply:
The `customServerCertificateId` is currently evaluated as part of the `equals(Object)` call. Does it need to be part of the `hashCode()` as well? What is criteria for inclusion? Also, why don't we rely on the default implementation of `hashCode()`?

Reply:
It is recommended that objects that are not equal return different hash codes. Not that it really matters here.